src/candidate.rs

```diff
@@ -559,9 +560,14 @@ impl Candidate {
         );
         Ok((e_tx, dt))
     }
+    pub(crate) fn with_orbit(&self, orbit: Orbit) -> Self {
```

Comment:
If `Candidate` can derive `Copy`, then this signature could be `fn with_orbit(mut self, orbit: Orbit)`. That's a syntax I'm ashamed to say I've only learned about recently (less than 6 months ago) despite years of Rust, ha!